Alignment/OfflineValidation/src/PlotAlignmentValidation.cc

```diff
@@ -1229,24 +1229,19 @@ void PlotAlignmentValidation::plotChi2(const char* inputFile) {
   // Opens the file (it should be OfflineValidation(Parallel)_result.root)
   // and reads and plots the norm_chi^2 and h_chi2Prob -distributions.
 
-  Bool_t errorflag = kTRUE;
+  Bool_t errorflag = kFALSE;
   TFile* fi1 = TFile::Open(inputFile, "read");
-  TDirectoryFile* mta1 = nullptr;
-  TDirectoryFile* mtb1 = nullptr;
-  TCanvas* normchi = nullptr;
-  TCanvas* chiprob = nullptr;
   if (fi1 != nullptr) {
-    mta1 = (TDirectoryFile*)fi1->Get("TrackerOfflineValidation");
-    if (mta1 != nullptr) {
-      mtb1 = (TDirectoryFile*)mta1->Get("GlobalTrackVariables");
-      if (mtb1 != nullptr) {
-        normchi = dynamic_cast<TCanvas*>(mtb1->Get("h_normchi2"));
-        chiprob = dynamic_cast<TCanvas*>(mtb1->Get("h_chi2Prob"));
-        if (normchi != nullptr && chiprob != nullptr) {
-          errorflag = kFALSE;
-        }
-      }
+    if (fi1->GetDirectory("TrackerOfflineValidation/GlobalTrackVariables") == nullptr) {
+      errorflag = kTRUE;
     }
+  } else {
+    errorflag = kTRUE;
+  }
+  TCanvas* normchi = new TCanvas((TCanvas*)fi1->Get("TrackerOfflineValidation/GlobalTrackVariables/h_normchi2"));
+  TCanvas* chiprob = new TCanvas((TCanvas*)fi1->Get("TrackerOfflineValidation/GlobalTrackVariables/h_chi2Prob"));
```

Comment:
```suggestion
  auto normchi = fi1->Get<TCanvas>("TrackerOfflineValidation/GlobalTrackVariables/h_normchi2"));
  auto chiprob = fi1->Get<TCanvas>("TrackerOfflineValidation/GlobalTrackVariables/h_chi2Prob"));
```

Reply:
Yes indeed, my initial solution was using wrong constructor. Thanks for spotting. I need to run in through the all tests again and will commit once more.

Reply:
```suggestion
  auto normchi = fi1->Get<TCanvas>("TrackerOfflineValidation/GlobalTrackVariables/h_normchi2"));
  normchi->GetFrame()->ResetBit(kCanDelete);
  auto chiprob = fi1->Get<TCanvas>("TrackerOfflineValidation/GlobalTrackVariables/h_chi2Prob"));
  chiprob->GetFrame()->ResetBit(kCanDelete);
```
is likely to be fixing the issue.

Reply:
I take back, still crashing.

Reply:
Ah great, I see your comment on bug in TPad https://github.com/cms-sw/cmssw/issues/40091#issuecomment-1320661264
It works with a workaround solution. I will make a comment to remove this extra step in this code with future ROOT version.